### ISSUE TITLE
ReparamModule Usage

### ISSUE THREAD

**Issue by Sinp17** [GeorgeCazenavette/mtt-distillation#30]
Hi George,

Thanks for your amazing work! 

I find a module called 'ReparamModule' in your code,  without too much explaination. It seems to flatten or unflatten parameters. I think its a supporting module for training of student models without using 'loss.backward()'. Could u give more explanation about this part?

Can I use this ReparamModule in language models like Transformers or Bert? Or which part should I change? I am a new grad student in NLP and RS, your answer would be a great help.

Much Thanks.

**Comment by GeorgeCazenavette** [comment]
Hi,

The ReparamModule is a wrapper for a nn.Module that allows it to take in the original input as well as the _parameters_ as inputs during the forward pass.

We used this so we could differentiate with respect to the parameter updates during the inner loop.

You could normally use some module like [higher](https://github.com/facebookresearch/higher) to get higher-order gradients, but we also needed gradients with respect to the learning rate.

The original repo for ReparamModule is here, written by Tongzhou (an author of this paper): https://github.com/ssnl/PyTorch-Reparam-Module

Hope this helps!

**Comment by Sinp17** [comment]
Thanks for your reply, I'll look into it carefully.
